lib/openfeature/sdk/configuration.rb

```diff
@@ -13,13 +13,22 @@ module SDK
     class Configuration
       extend Forwardable
 
-      attr_accessor :context, :provider, :hooks
+      attr_accessor :context, :hooks
+      attr_reader :provider
 
       def_delegator :@provider, :metadata
 
       def initialize
         @hooks = []
       end
+
+      def provider=(provider)
+        @provider.shutdown if @provider.respond_to?(:shutdown)
```

Comment:
Let's make sure that `@provider` is set before trying to shut it down. This work as expected when `@provider` hasn't been set because `nil` doesn't respond to `shutdown` , but checking `@provider` first would make it a lot more explicit and easier to read.

We could also use some specs around this shutdown behavior.

Reply:
I can add the explicit nil check if it makes clearer that @provider could not be set yet; would using safe navigation be more clear, i.e. `@provider&.respond_to?(:shutdown)`?

For the specs, the `"Requirement 1.1.2.3"` context in spec/specification/flag_evaluation_api_spec` was my attempt at testing the shutdown behavior. Are there more cases not tested there you can think of we should add to coverage?

Reply:
Going to wait one more day for @technicalpickles to take a look, then I'll resolve this conversation and merge in 🙏🏻